internal/node/internal-nats/internal_nats_server.go

```diff
@@ -213,7 +219,7 @@ func (s *InternalNatsServer) StoreFileForID(id string, bytes []byte) error {
 		return err
 	}
 
-	_, err = bucket.PutBytes(ctx, workloadCacheFileKey, bytes)
+	_, err = bucket.PutBytes(ctx, id, bytes)
```

Comment:
Note that this didn't need to be fixed. The single cache key worked fine because the cache bucket is inside an isolated account. Each agent has its own nkey which connects to its own isolated account.